src/analytics/analytics.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import { EventDocument, getCurrentEventState } from '@opencrvs/toolkit/events'
+import { birthEvent } from '@countryconfig/form/v2/birth'
+import { deathEvent } from '@countryconfig/form/v2/death'
+import { Event } from '@countryconfig/form/types/types'
+import { Kysely, sql } from 'kysely'
+import { compact as removeNulls } from 'lodash'
+
+interface AnalyticsEvent {
+  id: string
+
+  eventType: string
+  trackingId: string
+  declaration: Record<string, any>
+}
+
+export const upsertAnalyticsEvents = async (
+  values: Array<AnalyticsEvent>,
+  trx: Kysely<any>
```

Comment:
wonder if we want _some_ typing for Kysely, I guess we could now do it from the `AnalyticsEvent` interface

Reply:
Good enough for now even without typing imo